### PR TITLE
Refactor dom-accessor-wrapper into separate entities and rename it to code-instrumentation (close #11)

### LINKED ISSUE #11: Refactor dom-accessor-wrapper into separate entities and rename it to code-instrumentation.

### PR DESCRIPTION
**Pull request by churkin**
\cc @inikulin @LavrovArtem

### CODE REVIEW

File: src/client/hammerhead.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*eslint-disable no-native-reassign*/
 import * as Browser from './utils/browser';
 import * as DOM from './utils/dom';
-import { getAttributesProperty } from './sandboxes/dom-accessor-wrappers';
+import { getAttributesProperty } from './sandboxes/code-instrumentation/code-instrumentation';
```
Comment (inikulin) [inline src/client/hammerhead.js]: ./sandboxes/code-instrumentation/code-instrumentation -> ./sandboxes/code-instrumentation/index.js
So, import will turn into:

import { getAttributesProperty } from './sandboxes/code-instrumentation';

File: src/client/sandboxes/code-instrumentation/code-instrumentation.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import PropertiesInstrumentation from './properties/properties';
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/code-instrumentation.js]: Same as above

File: src/client/sandboxes/code-instrumentation/code-instrumentation.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import LocationInstrumentation from './location';
+import MethodsInstrumentation from './methods';
+import EventEmitter from '../../utils/event-emitter';
+import { getAttributeProperty } from './properties/attributes';
+import { PROCESS_SCRIPT_METH_NAME, process as processScript } from '../../../processing/js/index';
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/code-instrumentation.js]:
```js
import { PROCESS_SCRIPT_METH_NAME, process as processScript } from '../../../processing/js';
```

File: src/client/sandboxes/code-instrumentation/code-instrumentation.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    getAttributesProperty (el) {
+        return getAttributeProperty(el);
+    }
+
+    getOriginalErrorHandler (window) {
+        return PropertiesInstrumentation.getOriginalErrorHandler(window);
+    }
+
+    init (window, document) {
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/code-instrumentation.js]: If we have class we shouldn't have `init` method. All initialization should be done within ctor.

File: src/client/sandboxes/code-instrumentation/location.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
+import UrlUtil from '../../utils/url';
+import createPropertyDesc from '../../utils/create-property-desc.js';
+import { GET_LOCATION_METH_NAME, SET_LOCATION_METH_NAME } from '../../../processing/js/index';
+import { isLocation } from '../../utils/types';
+
+class LocationWrapper {
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/location.js]: It's inconsistent: `LocationWrapper` here and `LocationInstrumentation` above.

Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/location.js]: Oh, I see now. But shouldn't `LocationWrapper` and `LocationInstrumentation` be the same class?

File: src/client/sandboxes/code-instrumentation/location.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+    isLocationWrapper (obj) {
+        return obj instanceof LocationWrapper;
+    }
+
+    getLocationWrapper (owner) {
+        return owner[this.LOCATION_WRAPPER];
+    }
+
+    init (window, document) {
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/location.js]: Same here

File: src/client/sandboxes/code-instrumentation/code-instrumentation.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import PropertiesInstrumentation from './properties/properties';
+import LocationInstrumentation from './location';
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/code-instrumentation.js]: If it's an object and not class, it should be lower cased: `locationInstrumentation`

File: src/client/sandboxes/code-instrumentation/location.js (hunk 5 above)
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/location.js]: I get it now. Let's move classe to the separate files. Always try to follow the rule: one class - one file. So we will have:
`location/index.js`, `location/wrapper.js`

File: src/client/sandboxes/code-instrumentation/methods.js [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+                result.pop();
+
+                return result;
+            }
+        }
+
+        return args;
+    }
+
+    init (window) {
```
Comment (inikulin) [inline src/client/sandboxes/code-instrumentation/methods.js]: Same here

**Comment by inikulin** [comment]
\r-

**Comment by churkin** [comment]
FPR, tests have passed https://travis-ci.org/churkin/testcafe-hammerhead/jobs/78189132

**Comment by churkin** [comment]
@inikulin I still leave the 'codeInstrumentation.init' function name because all sandboxes have this function

**Comment by inikulin** [comment]
Well, it's done wrong. What's the purpose of classes then?

**Comment by inikulin** [comment]
Looks really good in general, just some minor naming issues.
\r-

**Comment by churkin** [comment]
@inikulin 
> Or even better DocumentLocationAccessorsInstrumentation

May be `WindowLocationAccessorsInstrumentation`? it is more correct.

**Comment by inikulin** [comment]
@churkin Looking at the spec now I think we should keep it just `LocationAccessorsInstrumentation`, since `Location` is linked to the both `window` and `document` and can be considered global.

**Comment by AlexanderMoskovkin** [comment]
the PR reopened because of testcafe-build-bot rebooting issue

**Comment by churkin** [comment]
FPR

**Comment by inikulin** [comment]
lgtm

**Comment by LavrovArtem** [comment]
lgtm

**Pull request merged**